src/nvidia_resiliency_ext/checkpointing/async_ckpt/core.py

```diff
@@ -546,10 +548,13 @@ def maybe_finalize_async_calls(self, blocking=False, no_dist=False) -> List[int]
                 call_idx, _, async_request = self.async_calls.popleft()
                 for finalize_fn in async_request.finalize_fns:
                     finalize_fn()
-                ten = torch.tensor([call_idx], dtype=torch.int, device=torch.cuda.current_device())
-                torch.distributed.all_reduce(ten, op=torch.distributed.ReduceOp.MAX)
-                assert ten.item() == call_idx, 'Unmatched async calls. '
-                'That probably means not all ranks are participating in async finalization'
+                if not no_dist:
```

Comment:
Thanks for fixing this. Double negative is usually not great. Can you change it to either `local` or `use_dist` and change the default and the condition accordingly?

Reply:
The variable exists in many different places for synchronization. I can rename it for a bit better readability but at some places, hard to avoid double negative.

Reply:
Let's keep this at this moment and change all use cases at a time later including examples.

Reply:
okay, sounds good